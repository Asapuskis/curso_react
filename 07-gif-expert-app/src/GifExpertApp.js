import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

  // Utilizaremos el hook useState para poder modificar los valores del array de categorías.
  const [categories, setCategories] = useState(defaultCategories);

  // Función para añadir nuevos elementos al array
  // const handleAdd = () => {
  //   // Las dos funciones hacen lo mismo
  //   // setCategories( (categories) => categories.concat('Demoni'));
  //   setCategories(cats => [...cats, 'Demoni']);
  // }

  return (
      <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories}/>
        <hr />
        <ol>
          {
            // map itera sobre un array y permite devolver el valor modificado de la forma que se quiera, en este caso
            // añadiendo las etiquetas <li></li> para añadirlas a la lista <ol></ol>
            categories.map( (category, index) => {
              return  <GifGrid 
                        category={category}
                        key={category}
                      />;
            })
          }
        </ol>
      </>
  )
}
