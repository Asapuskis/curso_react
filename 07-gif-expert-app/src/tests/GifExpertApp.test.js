import {shallow} from 'enzyme';
import { GifExpertApp } from "../GifExpertApp"

describe('Pruebas en <GifExpertApp />', () => {

    test('Debe mostrarse correctamente', () => { 
        const wrapper = shallow(<GifExpertApp />)
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe mostrar una lista de categorÃ­as', () => { 
        const categories = ['Shingeki', 'Sansfer'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        expect( wrapper ).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
     })
})