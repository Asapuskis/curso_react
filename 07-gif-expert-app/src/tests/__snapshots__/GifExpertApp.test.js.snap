// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Shingeki"
      key="Shingeki"
    />
    <GifGrid
      category="Sansfer"
      key="Sansfer"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
